aws/resource_aws_appconfig_configuration_profile.go

```diff
@@ -0,0 +1,301 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/appconfig"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+	iamwaiter "github.com/terraform-providers/terraform-provider-aws/aws/internal/service/iam/waiter"
+)
+
+func resourceAwsAppconfigConfigurationProfile() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsAppconfigConfigurationProfileCreate,
+		Read:   resourceAwsAppconfigConfigurationProfileRead,
+		Update: resourceAwsAppconfigConfigurationProfileUpdate,
+		Delete: resourceAwsAppconfigConfigurationProfileDelete,
+
+		Schema: map[string]*schema.Schema{
+			"application_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 64),
+				),
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(0, 1024),
+				),
+			},
+			"location_uri": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 2048),
+				),
+			},
+			"retrieval_role_arn": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateArn,
+			},
+			"validator": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{appconfig.ValidatorTypeJsonSchema, appconfig.ValidatorTypeLambda}, false),
```

Comment:
what happens when it isn't one of these types?

Reply:
There will be an error when Terraform validates the plan. This is good, since otherwise there would be an error during the apply when the config profile is being created.